cmd/channel/email/main.go

```diff
@@ -38,17 +44,49 @@ func (ch *Email) SendNotification(req *plugin.NotificationRequest) error {
 	}
 
 	var msg bytes.Buffer
-	_, _ = fmt.Fprintf(&msg, "To: %s\n", strings.Join(to, ","))
-	_, _ = fmt.Fprintf(&msg, "Subject: [#%d] %s %s is %s\n\n", req.Incident.Id, req.Event.Type, req.Object.Name, req.Event.Severity)
-
 	plugin.FormatMessage(&msg, req)
 
-	err := smtp.SendMail(ch.GetServer(), nil, ch.From, to, bytes.ReplaceAll(msg.Bytes(), []byte("\n"), []byte("\r\n")))
-	if err != nil {
-		return err
+	return enmime.Builder().
+		ToAddrs(to).
+		Subject(fmt.Sprintf("[#%d] %s %s is %s\n\n", req.Incident.Id, req.Event.Type, req.Object.Name, req.Event.Severity)).
+		From("", ch.From).
+		Text(msg.Bytes()).
+		Send(ch)
+}
+
+func (ch *Email) Send(reversePath string, recipients []string, msg []byte) error {
+	msgReader := bytes.NewReader(msg)
+	var auth sasl.Client
+	if ch.Password != "" {
+		auth = sasl.NewPlainClient("", ch.User, ch.Password)
 	}
 
-	return nil
+	switch ch.Encryption {
+	case "tls":
+		return smtp.SendMailTLS(ch.GetServer(), auth, reversePath, recipients, msgReader)
+	case "starttls":
+		return smtp.SendMail(ch.GetServer(), auth, reversePath, recipients, msgReader)
+	default:
+		c, err := smtp.Dial(ch.GetServer())
+		if err != nil {
+			return err
+		}
+		defer c.Close()
+
+		if auth != nil {
+			if ok, _ := c.Extension("AUTH"); !ok {
+				return errors.New("smtp: server doesn't support AUTH")
+			}
+			if err = c.Auth(auth); err != nil {
+				return err
+			}
+		}
+		if err := c.SendMail(reversePath, recipients, msgReader); err != nil {
+			return err
+		}
+
+		return c.Quit()
```

Comment:
However, instead of going with this home-brew approach, I would consider using a `smtp.Client` as I tried in <https://github.com/Icinga/icinga-notifications/commit/9308b97c2b615eb291e45bb317b9a997b7482866>. Please be aware that I have wrote this according to both go-smtp's API doc and source, but haven't tested it :-)